mavros_odom_sensor/src/mavros_odom_sensor.cpp

```diff
@@ -0,0 +1,157 @@
+
+  RCLCPP_INFO(  // NOLINT
+    rclcpp::get_logger("MavrosOdomSensor"), "Successfully initialized MavrosOdomSensor sensor interface!");
+
+  return hardware_interface::CallbackReturn::SUCCESS;
+}
+
+hardware_interface::CallbackReturn MavrosOdomSensor::on_configure(const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  RCLCPP_INFO(  // NOLINT
+    rclcpp::get_logger("MavrosOdomSensor"), "Configuring the MavrosOdomSensor sensor interface.");
+
+  mavros_odometry_topic_ = info_.hardware_parameters.at("mavros_odom_topic");
+  RCLCPP_INFO(
+    rclcpp::get_logger("MavrosOdomSensor"), "Subscribing to odometry topic: %s",
+    mavros_odometry_topic_.c_str());  // NOLINT
+
+  topic_based_odometry_subscriber_ = node_odom_->create_subscription<nav_msgs::msg::Odometry>(
+    mavros_odometry_topic_, rclcpp::SensorDataQoS(), [this](const nav_msgs::msg::Odometry::SharedPtr odometer_state) {
+      state_interface_config_.current_odom = *odometer_state;
+    });
+
+  return hardware_interface::CallbackReturn::SUCCESS;
+}
+
+hardware_interface::CallbackReturn MavrosOdomSensor::on_cleanup(const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  return hardware_interface::CallbackReturn::SUCCESS;
+}
+
+std::vector<hardware_interface::StateInterface> MavrosOdomSensor::export_state_interfaces()
+{
+  std::vector<hardware_interface::StateInterface> state_interfaces;
+
+  state_interfaces.emplace_back(
+    namespace_ + "joint_x", "position", &state_interface_config_.euler_current_odom.linear.x);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_y", "position", &state_interface_config_.euler_current_odom.linear.y);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_z", "position", &state_interface_config_.euler_current_odom.linear.z);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_rx", "position", &state_interface_config_.euler_current_odom.angular.x);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_ry", "position", &state_interface_config_.euler_current_odom.angular.y);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_rz", "position", &state_interface_config_.euler_current_odom.angular.z);
+
+  state_interfaces.emplace_back(
+    namespace_ + "joint_x", "velocity", &state_interface_config_.current_odom.twist.twist.linear.x);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_y", "velocity", &state_interface_config_.current_odom.twist.twist.linear.y);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_z", "velocity", &state_interface_config_.current_odom.twist.twist.linear.z);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_rx", "velocity", &state_interface_config_.current_odom.twist.twist.angular.x);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_ry", "velocity", &state_interface_config_.current_odom.twist.twist.angular.y);
+  state_interfaces.emplace_back(
+    namespace_ + "joint_rz", "velocity", &state_interface_config_.current_odom.twist.twist.angular.z);
```

Comment:
Could we rename these to `vx`, `vy`, `vz`, `wx`, `wy`, `wz`? Then it's clear what joint you are referring to and that it's a velocity without a verbose name.